### ISSUE TITLE
Feature: Support for dependency injection with Quarkus

### ISSUE THREAD

**Issue by bnjns** [mockk/mockk#552]
### Prerequisites

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Expected Behavior

We currently use [Quarkus](https://quarkus.io) as our framework; this has an extension that provides Mockito support (`io.quarkus:quarkus-junit5-mockito`) but I'd also love to be able to use MockK due to the significantly better syntax.

With Quarkus you can inject a mocked or spied bean using `@InjectMock` and `@InjectSpy`, which act similarly to Spring's `@MockBean` and `@SpyBean`.

Is there a way to be able to use MockK for these dependencies? Currently I just get a variety of errors, presumably because the injected beans aren't created by MockK?

For example:

```kotlin
@QuarkusTest
class ExampleTest {
	@Inject
	lateinit var service: Service

	@InjectMock
	lateinit var mock: MockedBean

	@InjectSpy
	lateinit var spy: SpiedBean

	@Test
	fun exampleTest() {
		every { mock.test() } returns "Hello"
        
        service.execute()
   		
		verify(exactly = 1) {
			spy.called()
		}
	}
}
```

The above would give me the following error

```
io.mockk.MockKException: Missing calls inside every { ... } block.
```

### Current behavior

I _can_ make it work by manually creating the beans I wish to mock/spy and then manually adding them into Quarkus' DI container, but it's not pretty (especially if any of these have their own dependencies and this is technically a deprecated method of mocking/spying with Quarkus).

```kotlin
@QuarkusTest
class ExampleTest {
	@Inject
	lateinit var service: Service

	@Inject
	lateinit var spyDep: SpiedBeanDependency

	private val mock: MockedBean = mockk()
	private lateinit var spy: SpiedBean

	@BeforeAll
	fun configureMocks() {
		spy = spyk(SpiedBean(spyDep))
		QuarkusMock.installMockForType(spy, SpiedBean::class.java)
		QuarkusMock.installMockForType(mock, MockedBean::class.java)
	}

	@Test
	fun exampleTest() {
		every { mock.test() } returns "Hello"
        
        service.execute()
   		
		verify(exactly = 1) {
			spy.called()
		}
	}
}
```

**Comment by Raibaz** [comment]
I think dealing with DI frameworks is, at least for the time being, outside of the scope of MockK. For instance, for Spring boot there is a separate library, [springmockk](https://github.com/Ninja-Squad/springmockk), that replaces the usual Spring boot test annotations with mockk annotations.

Random question, though: have you tried annotating your dependencies with _both_ `@Inject` and `@MockK` and calling `MockkAnnotations.init(this)` in your `@BeforeAll` function? I'm guessing it won't work but it may be worth giving a shot.

**Comment by bnjns** [comment]
@Raibaz Yeah that's fair enough - it was a long shot!

I did try using both annotations, using `@ExtendsWith` rather than `MockkAnnotations.init(this)`, but I ended up getting an `UninitializedPropertyAccessException`. I'll give `MockkAnnotations.init(this)` a go but I suspect it'll give me the same result.

I might try and see if I can make something similar to springmockk for Quarkus to tie the 2 together.
